### PR TITLE
Add PodUtils tweaks for testing

### PR DESCRIPTION
**Pull request by kornys**
## Description

Add PodUtils with common tweaks methods for verify pod stability, state etc...

## Type of Change

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit/integration tests pass locally with my changes

### CODE REVIEW

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+                                            || cs.getState().getTerminated().getReason().equals("Completed"))) {
+                                        LOGGER.debug("Container: {} of Pod: {}/{} not ready",
+                                                namespaceName, pod.getMetadata().getName(), cs.getName());
+                                        return false;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    LOGGER.info("Pods in namespace {} are ready", namespaceName);
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:68]:
```suggestion
                    LOGGER.info("All pods in namespace {} are ready", namespaceName);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+     * @param onTimeout callback on timeout
+     */
+    public static void waitForPodsReady(String namespaceName, boolean containers, Runnable onTimeout) {
+        Wait.until("readiness of all Pods matching in namespace " + namespaceName,
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_MEDIUM, READINESS_TIMEOUT,
+                () -> {
+                    List<Pod> pods = KubeResourceManager.getKubeClient().getClient()
+                            .pods().inNamespace(namespaceName).list().getItems();
+                    if (pods.isEmpty()) {
+                        LOGGER.debug("Expected Pods are not ready!");
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:48]:
```suggestion
                        LOGGER.debug("There are no existing pods in namespace {}", namespaceName);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    /**
+     * Wait for all pods in namespace to be ready
+     *
+     * @param namespaceName name of the namespace
+     * @param containers flag wait for all containers
+     * @param onTimeout callback on timeout
+     */
+    public static void waitForPodsReady(String namespaceName, boolean containers, Runnable onTimeout) {
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:41]:
```suggestion
    public static void waitForPodsReady(String namespaceName, boolean containersReady, Runnable onTimeout) {
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+                () -> {
+                    List<Pod> pods = KubeResourceManager.getKubeClient().getClient()
+                            .pods().inNamespace(namespaceName).list().getItems();
+                    if (pods.isEmpty()) {
+                        LOGGER.debug("Expected Pods are not ready!");
+                        return false;
+                    }
+                    for (Pod pod : pods) {
+                        if (!(Readiness.isPodReady(pod) || Readiness.isPodSucceeded(pod))) {
+                            LOGGER.debug("Pod not ready: {}/{}", namespaceName, pod.getMetadata().getName());
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:53]:
```suggestion
                            LOGGER.debug("There is not ready pod {}/{}", namespaceName, pod.getMetadata().getName());
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+                            return false;
+                        } else {
+                            if (containers) {
+                                for (ContainerStatus cs : pod.getStatus().getContainerStatuses()) {
+                                    if (!(Boolean.TRUE.equals(cs.getReady())
+                                            || cs.getState().getTerminated().getReason().equals("Completed"))) {
+                                        LOGGER.debug("Container: {} of Pod: {}/{} not ready",
+                                                namespaceName, pod.getMetadata().getName(), cs.getName());
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:61]:
```suggestion
                                        LOGGER.debug("Container {} of Pod {}/{} is not ready",
                                                cs.getName(), namespaceName, pod.getMetadata().getName());
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+     */
+    public static void waitForPodsReady(String namespaceName, LabelSelector selector, int expectPods,
+                                        boolean containers, Runnable onTimeout) {
+        Wait.until("readiness of all Pods matching: " + selector,
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_MEDIUM, READINESS_TIMEOUT,
+                () -> {
+                    List<Pod> pods = KubeResourceManager.getKubeClient().getClient().pods()
+                            .inNamespace(namespaceName).withLabelSelector(selector).list().getItems();
+                    if (pods.isEmpty() && expectPods == 0) {
+                        LOGGER.debug("Expected Pods are ready");
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:90]:
```suggestion
                        LOGGER.info("All expected Pods {} in namespace {} are ready", selector, namesapceName);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    /**
+     * Wait for pods selected by label selector in namespace to be ready
+     *
+     * @param namespaceName namespace
+     * @param selector label selector of the pods
+     * @param expectPods expected pods count
+     * @param containers flag wait for all containers
+     * @param onTimeout callback on timeout
+     */
+    public static void waitForPodsReady(String namespaceName, LabelSelector selector, int expectPods,
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:82]:
```suggestion
    public static void waitForPodsReady(String namespaceName, LabelSelector selector, int expectPodsCount,
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+                        LOGGER.debug("Pods matching: {}/{} are not ready", namespaceName, selector);
+                        return false;
+                    }
+                    if (pods.size() != expectPods) {
+                        LOGGER.debug("Expected Pods: {}/{} are not ready", namespaceName, selector);
+                        return false;
+                    }
+                    for (Pod pod : pods) {
+                        if (!(Readiness.isPodReady(pod) || Readiness.isPodSucceeded(pod))) {
+                            LOGGER.debug("Pod not ready: {}/{}", namespaceName, pod.getMetadata().getName());
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:103]:
```suggestion
                            LOGGER.debug("Pod is not ready: {}/{}", namespaceName, pod.getMetadata().getName());
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+                            return false;
+                        } else {
+                            if (containers) {
+                                for (ContainerStatus cs : pod.getStatus().getContainerStatuses()) {
+                                    if (!(Boolean.TRUE.equals(cs.getReady())
+                                            || cs.getState().getTerminated().getReason().equals("Completed"))) {
+                                        LOGGER.debug("Container: {} of Pod: {}/{} not ready",
+                                                namespaceName, pod.getMetadata().getName(), cs.getName());
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:111]:
```suggestion
                                        LOGGER.debug("Container {} of Pod {}/{} not ready",
                                                pod.getMetadata().getName(), cs.getName(), namespaceName);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_MEDIUM, READINESS_TIMEOUT,
+                () -> {
+                    List<Pod> pods = KubeResourceManager.getKubeClient().getClient().pods()
+                            .inNamespace(namespaceName).withLabelSelector(selector).list().getItems();
+                    if (pods.isEmpty() && expectPods == 0) {
+                        LOGGER.debug("Expected Pods are ready");
+                        return true;
+                    }
+                    if (pods.isEmpty()) {
+                        LOGGER.debug("Pods matching: {}/{} are not ready", namespaceName, selector);
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:94]:
```suggestion
                        LOGGER.debug("Pods matching {}/{} are not ready", namespaceName, selector);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+                        return false;
+                    }
+                    if (pods.size() != expectPods) {
+                        LOGGER.debug("Expected Pods: {}/{} are not ready", namespaceName, selector);
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:98]:
```suggestion
                        LOGGER.debug("Expected Pods {}/{} are not ready", namespaceName, selector);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+                                            || cs.getState().getTerminated().getReason().equals("Completed"))) {
+                                        LOGGER.debug("Container: {} of Pod: {}/{} not ready",
+                                                namespaceName, pod.getMetadata().getName(), cs.getName());
+                                        return false;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    LOGGER.info("Pods matching: {}/{} are ready", namespaceName, selector);
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:118]:
```suggestion
                    LOGGER.info("Pods matching {}/{} are ready", namespaceName, selector);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+     * @param expectedPods expected pods count
+     * @param containers flag wait for containers
+     */
+    public static void waitForPodsReadyWithRestart(String namespace, LabelSelector selector,
+                                                   int expectedPods, boolean containers) {
+        try {
+            waitForPodsReady(namespace, selector, expectedPods, containers, () -> {
+            });
+        } catch (Exception ex) {
+            LOGGER.warn("Pods not ready trying to restart");
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:137]:
```suggestion
            LOGGER.warn("Pods {}/{} are not ready. Going to restart them", namespaceName, selector);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+
+    /**
+     * Wait for pod ready, if not ready for timeout try to restart and check again
+     *
+     * @param namespace namespace
+     * @param selector label selector
+     * @param expectedPods expected pods count
+     * @param containers flag wait for containers
+     */
+    public static void waitForPodsReadyWithRestart(String namespace, LabelSelector selector,
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:131]:
```suggestion
    public static void waitForPodsReadyWithRestart(String namespaceName, LabelSelector selector,
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+                                                   int expectedPods, boolean containers) {
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:132]:
```suggestion
                                                   int expectedPodsCount, boolean containersReady) {
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+     * Wait for pods selected by label selector in namespace to be ready
+     *
+     * @param namespaceName namespace
+     * @param selector label selector of the pods
+     * @param expectPods expected pods count
+     * @param containers flag wait for all containers
+     * @param onTimeout callback on timeout
+     */
+    public static void waitForPodsReady(String namespaceName, LabelSelector selector, int expectPods,
+                                        boolean containers, Runnable onTimeout) {
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:83]:
```suggestion
                                        boolean containersReady, Runnable onTimeout) {
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+                                pod -> pod.getMetadata().getUid()));
+    }
+
+    /**
+     * Verify if the pod is stable after it is in ready state.
+     *
+     * @param namespaceName namespace
+     * @param labelSelector label selector
+     */
+    public static void verifyThatPodsAreStable(String namespaceName, LabelSelector labelSelector) {
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:169]:
```suggestion
    public static void verifyThatPodsAreStable(String namespaceName, LabelSelector selector) {
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+        int[] stabilityCounter = {0};
+        String phase = "Running";
+
+        Wait.until(String.format("Pods in namespace '%s' with LabelSelector %s stability in phase %s",
+                        namespaceName, labelSelector, phase),
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_SHORT, TestFrameConstants.GLOBAL_TIMEOUT,
+                () -> {
+                    List<Pod> existingPod = KubeResourceManager.getKubeClient().getClient().pods()
+                            .inNamespace(namespaceName).withLabelSelector(labelSelector).list().getItems();
+                    LOGGER.debug("Working with the following pods: {}", existingPod.stream()
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:179]:
```suggestion
                    LOGGER.debug("Considering the following pods {}", existingPod.stream()
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 19]
```diff
@@ -0,0 +1,215 @@
...
+                    for (Pod pod : existingPod) {
+                        if (pod == null) {
+                            continue;
+                        }
+                        if (pod.getStatus().getPhase().equals(phase)) {
+                            LOGGER.debug("Pod: {}/{} is in the {} state. " +
+                                            "Remaining seconds for Pod to be stable {}",
+                                    namespaceName,
+                                    pod.getMetadata().getName(),
+                                    pod.getStatus().getPhase(),
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:191]:
```suggestion
                            LOGGER.debug("Pod {}/{} is in the {} state. " +
                                            "Remaining seconds for Pod to be stable {}",
                                    namespaceName,
                                    pod.getMetadata().getName(),
                                    pod.getStatus().getPhase(),
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 20]
```diff
@@ -0,0 +1,215 @@
...
+                                    TestFrameConstants.GLOBAL_STABILITY_TIME -
+                                            (TestFrameConstants.GLOBAL_POLL_INTERVAL_SHORT * stabilityCounter[0])
+                            );
+                        } else {
+                            LOGGER.warn("Pod: {}/{} is not stable in phase following phase {} ({})" +
+                                            " reset the stability counter from {}s to {}s",
+                                    namespaceName, pod.getMetadata().getName(), pod.getStatus().getPhase(),
+                                    phase, stabilityCounter[0], 0);
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:199]:
```suggestion
                            LOGGER.warn("Pod {}/{} is not stable in phase following phase {} ({})" +
                                            " reset the stability counter from {}s to {}s",
                                    namespaceName, pod.getMetadata().getName(), pod.getStatus().getPhase(),
                                    phase, stabilityCounter[0], 0);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 21]
```diff
@@ -0,0 +1,215 @@
...
+                            stabilityCounter[0] = 0;
+                            return false;
+                        }
+                    }
+                    stabilityCounter[0]++;
+
+                    if (stabilityCounter[0] == TestFrameConstants.GLOBAL_STABILITY_TIME /
+                            (TestFrameConstants.GLOBAL_POLL_INTERVAL_SHORT)) {
+                        LOGGER.info("All Pods are stable {}", existingPod.stream()
+                                .map(p -> p.getMetadata().getName()).collect(Collectors.joining(" ,")));
```
Comment (Frawless) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:209]:
```suggestion
                        LOGGER.info("All Pods {}/{} are stable", namespaceName, existingPod.stream()
                                .map(p -> p.getMetadata().getName()).collect(Collectors.joining(" ,")));
```

**Review by Frawless — APPROVED** [review]

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 22]
```diff
@@ -0,0 +1,215 @@
...
+
+    /**
+     * Wait for all pods in namespace to be ready
+     *
+     * @param namespaceName name of the namespace
+     * @param containersReady flag wait for all containers
+     * @param onTimeout callback on timeout
+     */
+    public static void waitForPodsReady(String namespaceName, boolean containersReady, Runnable onTimeout) {
+        Wait.until("readiness of all Pods matching in namespace " + namespaceName,
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:42]: all Pods matching what?

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 23]
```diff
@@ -0,0 +1,215 @@
...
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_MEDIUM, READINESS_TIMEOUT,
+                () -> {
+                    List<Pod> pods = KubeResourceManager.getKubeClient().getClient()
+                            .pods().inNamespace(namespaceName).list().getItems();
+                    if (pods.isEmpty()) {
+                        LOGGER.debug("There are no existing pods in namespace {}", namespaceName);
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:48]:
```suggestion
                        LOGGER.debug("There are no existing Pods in Namespace: {}", namespaceName);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 24]
```diff
@@ -0,0 +1,215 @@
...
+                        return false;
+                    }
+                    for (Pod pod : pods) {
+                        if (!(Readiness.isPodReady(pod) || Readiness.isPodSucceeded(pod))) {
+                            LOGGER.debug("There is not ready pod {}/{}", namespaceName, pod.getMetadata().getName());
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:53]:
```suggestion
                            LOGGER.debug("There is not ready Pod {}/{}", namespaceName, pod.getMetadata().getName());
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 25]
```diff
@@ -0,0 +1,215 @@
...
+                                            || cs.getState().getTerminated().getReason().equals("Completed"))) {
+                                        LOGGER.debug("Container {} of Pod {}/{} is not ready",
+                                                cs.getName(), namespaceName, pod.getMetadata().getName());
+                                        return false;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    LOGGER.info("All pods in namespace {} are ready", namespaceName);
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:68]:
```suggestion
                    LOGGER.info("All Pods in Namespace: {} are ready", namespaceName);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 26]
```diff
@@ -0,0 +1,215 @@
...
+     *
+     * @param namespaceName namespace
+     * @param selector label selector of the pods
+     * @param expectPodsCount expected pods count
+     * @param containers flag wait for all containers
+     * @param onTimeout callback on timeout
+     */
+    public static void waitForPodsReady(String namespaceName, LabelSelector selector, int expectPodsCount,
+                                        boolean containers, Runnable onTimeout) {
+        Wait.until("readiness of all Pods matching: " + selector,
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:84]: mention something like `in Namespace: {}`

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 27]
```diff
@@ -0,0 +1,215 @@
...
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_MEDIUM, READINESS_TIMEOUT,
+                () -> {
+                    List<Pod> pods = KubeResourceManager.getKubeClient().getClient().pods()
+                            .inNamespace(namespaceName).withLabelSelector(selector).list().getItems();
+                    if (pods.isEmpty() && expectPodsCount == 0) {
+                        LOGGER.debug("All expected Pods {} in namespace {} are ready", selector, namespaceName);
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:90]:
```suggestion
                        LOGGER.debug("All expected Pods {} in Namespace: {} are ready", selector, namespaceName);
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 28]
```diff
@@ -0,0 +1,215 @@
...
+     * @param namespaceName namespace
+     * @param selector label selector
+     * @param expectedPodsCount expected pods count
+     * @param containersReady flag wait for containers
+     */
+    public static void waitForPodsReadyWithRestart(String namespaceName, LabelSelector selector,
+                                                   int expectedPodsCount, boolean containersReady) {
+        try {
+            waitForPodsReady(namespaceName, selector, expectedPodsCount, containersReady, () -> {
+            });
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:135]:
```suggestion
            waitForPodsReady(namespaceName, selector, expectedPodsCount, containersReady, () -> {});
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 29]
```diff
@@ -0,0 +1,215 @@
...
+        } catch (Exception ex) {
+            LOGGER.warn("Pods {}/{} are not ready. Going to restart them", namespaceName, selector);
+            KubeResourceManager.getKubeClient().getClient().pods()
+                    .inNamespace(namespaceName).withLabelSelector(selector).list().getItems().forEach(p ->
+                            KubeResourceManager.getKubeClient().getClient().resource(p).delete());
+            waitForPodsReady(namespaceName, selector, expectedPodsCount, containersReady, () -> {
+            });
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:142]:
```suggestion
            waitForPodsReady(namespaceName, selector, expectedPodsCount, containersReady, () -> {});
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 30]
```diff
@@ -0,0 +1,215 @@
...
+     * Wait for pod ready, if not ready for timeout try to restart and check again
+     *
+     * @param namespaceName namespace
+     * @param selector label selector
+     * @param expectedPodsCount expected pods count
+     * @param containersReady flag wait for containers
+     */
+    public static void waitForPodsReadyWithRestart(String namespaceName, LabelSelector selector,
+                                                   int expectedPodsCount, boolean containersReady) {
+        try {
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:133]: Why is try catch here?

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 31]
```diff
@@ -0,0 +1,215 @@
...
+     * Verify if the pod is stable after it is in ready state.
+     *
+     * @param namespaceName namespace
+     * @param selector label selector
+     */
+    public static void verifyThatPodsAreStable(String namespaceName, LabelSelector selector) {
+        int[] stabilityCounter = {0};
+        String phase = "Running";
+
+        Wait.until(String.format("Pods in namespace '%s' with LabelSelector %s stability in phase %s",
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:173]:
```suggestion
        Wait.until(String.format("Pods in Namespace: %s with LabelSelector: %s stability in phase: %s",
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java [hunk 32]
```diff
@@ -0,0 +1,215 @@
...
+                        namespaceName, selector, phase),
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_SHORT, TestFrameConstants.GLOBAL_TIMEOUT,
+                () -> {
+                    List<Pod> existingPod = KubeResourceManager.getKubeClient().getClient().pods()
+                            .inNamespace(namespaceName).withLabelSelector(selector).list().getItems();
+                    LOGGER.debug("Considering the following pods {}", existingPod.stream()
```
Comment (im-konge) [inline test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:179]:
```suggestion
                    LOGGER.debug("Considering the following Pods: {}", existingPod.stream()
```

File: test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java (hunk 30 above)
Comment (kornys) [reply test-frame-common/src/main/java/io/skodjob/testframe/utils/PodUtils.java:133]: Wait exception is thrown so it is correct... as wait exception is based on runtime ex which is not defined in method header....

**Review by im-konge — APPROVED** [review]

**Pull request merged**